android/android_browser_tests.gni

```diff
@@ -0,0 +1,256 @@
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_context_helper.h",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_context_util.cc",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_context_util.h",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_contribution.cc",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_contribution.h",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_network_util.cc",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_network_util.h",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_promotion.cc",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_promotion.h",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_response.cc",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_response.h",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_util.cc",
+  "//brave/components/brave_rewards/browser/test/common/rewards_browsertest_util.h",
+  "//brave/components/brave_rewards/browser/test/rewards_browsertest.cc",
+  "//brave/components/brave_rewards/browser/test/rewards_contribution_browsertest.cc",
+  "//brave/components/brave_rewards/browser/test/rewards_flag_browsertest.cc",
+  "//brave/components/brave_rewards/browser/test/rewards_notification_browsertest.cc",
+  "//brave/components/brave_rewards/browser/test/rewards_p3a_browsertest.cc",
+  "//brave/components/brave_rewards/browser/test/rewards_promotion_browsertest.cc",
+  "//brave/components/brave_rewards/browser/test/rewards_publisher_browsertest.cc",
+  "//brave/components/brave_rewards/browser/test/rewards_state_browsertest.cc",
+  "//brave/components/brave_shields/browser/https_everywhere_service_browsertest.cc",
+  "//brave/components/brave_shields/browser/test_filters_provider.cc",
+  "//brave/components/content_settings/renderer/brave_content_settings_agent_impl_browsertest.cc",
+  "//brave/third_party/blink/renderer/modules/brave/navigator_browsertest.cc",
+  "//chrome/browser/extensions/browsertest_util.cc",
+  "//chrome/browser/extensions/browsertest_util.h",
+  "//chrome/browser/extensions/extension_browsertest.cc",
+  "//chrome/browser/extensions/extension_browsertest.h",
+  "//chrome/browser/extensions/extension_function_test_utils.cc",
+  "//chrome/browser/extensions/extension_function_test_utils.h",
+  "//chrome/browser/media/media_browsertest.cc",
+  "//chrome/browser/media/media_browsertest.h",
+  "//chrome/browser/ui/test/test_browser_dialog.cc",
+  "//chrome/browser/ui/test/test_browser_dialog.h",
+  "//chrome/browser/ui/test/test_browser_ui.cc",
+  "//chrome/browser/ui/test/test_browser_ui.h",
+]
+
+android_test_exception_deps = [
+  "//brave/browser/autoplay:browser_tests",
```

Comment:
I think it's going to be too confusing to try to manage non _tests deps here. Let's just keep the `_tests` entries in this list

Reply:
I removed most deps that are not `_tests`. There are several deps that have to be excluded here because they include `gn` files start with some kind of Assert of flags which is off on Android.